src/components/carousel/_index.scss

```
@@ -0,0 +1,54 @@
+@use "../../scss";
+
+.c-carousel {
+	--slide-width: var(--carousel-slide-width, 25%);
+
+	display: grid;
+	grid-template-areas: "carousel";
+	justify-content: center;
+	overflow: hidden;
+
+	@include scss.component-properties("carousel");
+
+	> * {
+		grid-area: carousel;
+	}
+
+	&__track {
+		display: flex;
+		transition: transform 0.4s ease-in;
+		will-change: transform;
+
+		@include scss.component-properties("carousel-track");
+	}
+
+	&__slide {
+		flex: 0 0 var(--slide-width);
+
+		@include scss.component-properties("carousel-slide");
+	}
+
+	&__actions {
+		display: flex;
+		justify-content: space-between;
+		place-self: center;
+		width: 100%;
+		z-index: 1;
+
+		@include scss.component-properties("carousel-actions");
+	}
+
+	&__button {
+		@include scss.component-properties("carousel-button");
```

Part 3 of touch fix:
```suggestion
		z-index: 1;
		pointer-events: none;

		@include scss.component-properties("carousel-actions");
	}

	&__button {
		pointer-events: auto;

		@include scss.component-properties("carousel-button");
```